code/mob/living/carbon/human.dm

```diff
@@ -3149,8 +3149,18 @@
 
 	if (head)
 		worn += head.name
-	else
-		worn += "hair"
+	else if (!src.mutantrace || istype(src.mutantrace, /datum/mutantrace/dwarf) || istype(src.mutantrace, /datum/mutantrace/flashy))
+		if ((src.bioHolder.mobAppearance.customization_first in hair_styles) || (src.bioHolder.mobAppearance.customization_second in hair_styles) || (src.bioHolder.mobAppearance.customization_third in hair_styles))
+			worn += "hair"
+		else
+			worn += "bald head"
+		if ((src.bioHolder.mobAppearance.customization_first in moustache_styles) || (src.bioHolder.mobAppearance.customization_second in moustache_styles) || (src.bioHolder.mobAppearance.customization_third in moustache_styles))
+			worn += "moustache"
+		if ((src.bioHolder.mobAppearance.customization_first in beard_styles) || (src.bioHolder.mobAppearance.customization_second in beard_styles) || (src.bioHolder.mobAppearance.customization_third in beard_styles))
```

Comment:
Sorry, but I had another look at the code and this if-block is also problematic. This is doing much too much checks for it to be run every time someone self-clicks. Worst case its looping through those 3 lists (which have like 150 elements in total), 3 times each. It's too much to add for flavor text. 

I think this would be good if you could condense these into just one or 3 checks, which you could do if each hairstyle was a datum that had different attributes like being head hair, facial hair, hair length, and whatnot. It's kinda too much going on for just flavor text.